unstructured_ingest/v2/pipeline/pipeline.py

```diff
@@ -228,26 +231,42 @@ def _run(self):
                 logger.info("No files to process after filtering uncompressed content, exiting")
                 return
 
-        if not downloaded_data:
+        if not downloaded_data or self.context.download_only:
             return
 
         # Partition content
         elements = self.partitioner_step(downloaded_data)
+        # Download data non longer needed, delete if possible
```

Comment:
I don't think this respects --download-dir

Reply:
There's something about the combination of --work-dir and these new flags that is not working. If --work-dir is specified and --download-dir is not, I don't see the downloads ever in the work dir. I think it saves them in the .cache directory.

Reply:
```
artition step finished in 9.000297s
2024-09-13 11:25:29,894 MainProcess INFO     deleting download dir /Users/potter/Documents/unstructured-ingest/test_e2e/workdir/s3/download
2024-09-13 11:25:29,895 MainProcess INFO     ingest process finished in 15.046469s
2024-09-13 11:25:29,895 MainProcess INFO     deleting cache directory: /Users/potter/Documents/unstructured-ingest/test_e2e/workdir/s3
2024-09-13 11:25:29,898 MainProcess ERROR    failed to run source command s3: [Errno 2] No such file or directory: '/Users/potter/Documents/unstructured-ingest/test_e2e/workdir/s3/download'
Traceback (most recent call last):
  File "/Users/potter/Documents/unstructured-ingest/unstructured_ingest/v2/cli/base/src.py", line 59, in cmd
    pipeline.run()
  File "/Users/potter/Documents/unstructured-ingest/unstructured_ingest/v2/pipeline/pipeline.py", line 137, in run
    self._run()
  File "/Users/potter/Documents/unstructured-ingest/unstructured_ingest/v2/pipeline/pipeline.py", line 243, in _run
    shutil.rmtree(cache_dir)
  File "/Users/potter/.pyenv/versions/3.10.13/lib/python3.10/shutil.py", line 715, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/Users/potter/.pyenv/versions/3.10.13/lib/python3.10/shutil.py", line 713, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/potter/Documents/unstructured-ingest/test_e2e/workdir/s3/download'
Error: [Errno 2] No such file or directory: '/Users/potter/Documents/unstructured-ingest/test_e2e/workdir/s3/download'

Reply:
Good point, I had the download step return the download directory location for it's cache dir so it should work with this setup.